tests/unit/s2n_early_data_test.c

```diff
@@ -793,5 +793,188 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(conn));
     }
 
+    /* Test s2n_early_data_record_bytes */
+    {
+        /* Safety check */
+        EXPECT_ERROR_WITH_ERRNO(s2n_early_data_record_bytes(NULL, 1), S2N_ERR_NULL);
+
+        const uint32_t limit = 10;
+
+        struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(conn);
+        EXPECT_OK(s2n_append_test_psk_with_early_data(conn, limit, &s2n_tls13_aes_256_gcm_sha384));
+
+        /* If early data not expected, bytes are not recorded */
+        conn->early_data_bytes = 0;
+        EXPECT_OK(s2n_early_data_record_bytes(conn, limit + 1));
+        EXPECT_EQUAL(conn->early_data_bytes, 0);
+
+        EXPECT_SUCCESS(s2n_connection_set_early_data_expected(conn));
+
+        /* Bytes are recorded */
+        {
+            conn->early_data_bytes = 0;
+            EXPECT_OK(s2n_early_data_record_bytes(conn, 1));
+            EXPECT_EQUAL(conn->early_data_bytes, 1);
+
+            conn->early_data_bytes = 1;
+            EXPECT_OK(s2n_early_data_record_bytes(conn, 1));
+            EXPECT_EQUAL(conn->early_data_bytes, 2);
+        }
+
+        /* Error if exceeds max_early_data_size */
+        {
+            conn->early_data_bytes = 0;
+            EXPECT_OK(s2n_early_data_record_bytes(conn, limit));
+            EXPECT_EQUAL(conn->early_data_bytes, limit);
+
+            conn->early_data_bytes = 1;
+            EXPECT_ERROR_WITH_ERRNO(s2n_early_data_record_bytes(conn, limit), S2N_ERR_MAX_EARLY_DATA_SIZE);
+            EXPECT_EQUAL(conn->early_data_bytes, limit + 1);
+        }
+
+        /* Bytes not record if would overflow */
```

Comment:
Not really sure what this comment is trying to say. Seems like the behavior is that we do record bytes even if we go over the limit.

Reply:
Would "early_data_bytes can't overflow" make more sense? We might record /some/ bytes, but we stop completely when we reach UINT64_MAX.

Reply:
yeah that works for me